scipy/optimize/_qap.py

```diff
@@ -0,0 +1,437 @@
+import numpy as np
+import math
+from . import linear_sum_assignment, minimize_scalar, OptimizeResult
+
+
+def quadratic_assignment(
+    cost_matrix,
+    dist_matrix,
+    seed=None,
+    maximize=False,
+    n_init=1,
+    init="barycenter",
+    maxiter=30,
+    shuffle_input=True,
+    eps=0.05,
+):
+    r"""
+    Solve the quadratic assignment problem.
+
+    This function solves the Quadratic Assignment Problem (QAP) and the
+    Graph Matching Problem through an implementation of the Fast
+    Approximate QAP Algorithm (FAQ) (these two problems are the same up
+    to a sign change) [1]_.
+
+    Quadratic Assignment solves problems of the following form:
+
+    .. math::
+
+        \min_P & \ {-\text{trace}(APB^T P^T)}\\
```

Comment:
@asaadeldin11 this is opposite of QAP as defined in https://en.wikipedia.org/wiki/Quadratic_assignment_problem
aka this is the objective func for graph matching problem, right? Regardless we should make it clear which for the docs

Reply:
yes you're right that's the GMP objective function. Thanks for pointing it out I'll fix it